scipy/sparse/_base.py

```diff
@@ -328,30 +328,35 @@ def imag(self):
     def __repr__(self):
         _, format_name = _formats[self.format]
         sparse_cls = 'array' if isinstance(self, sparray) else 'matrix'
-        shape_str = 'x'.join(str(x) for x in self.shape)
         return (
-            f"<{shape_str} sparse {sparse_cls} of type '{self.dtype.type}'\n"
-            f"\twith {self.nnz} stored elements in {format_name} format>"
+            f"<{format_name} sparse {sparse_cls} of dtype '{self.dtype}'\n"
+            f"\twith {self.nnz} stored elements and shape {self.shape}>"
         )
 
     def __str__(self):
         maxprint = self._getmaxprint()
 
         A = self.tocoo()
 
+
         # helper function, outputs "(i,j)  v"
-        def tostr(row, col, data):
-            triples = zip(list(zip(row, col)), data)
+        def tostr(coords, data):
+            triples = zip(list(zip(*coords)), data)
             return '\n'.join([('  {}\t{}'.format(*t)) for t in triples])
```

Comment:
Technically these aren't really triples, they're (idx, value) pairs where `idx` is a tuple of coordinates.

How about this?

```python
pairs = zip(zip(*coords), data)
return '\n'.join(f'  {idx}\t{val}' for idx, val in pairs)
```